CRM/Mailing/BAO/Mailing.php

```diff
@@ -109,46 +109,227 @@ public static function getRecipientsCount($job_id, $mailing_id = NULL) {
   }
 
   /**
-   * note that $job_id is used only as a variable in the temp table construction
-   * and does not play a role in the queries generated.
-   * @param int $job_id
-   *   (misnomer) a nonce value used to name temporary tables.
-   * @param int $mailing_id
-   * @param bool $storeRecipients
-   * @param bool $dedupeEmail
-   * @param null $mode
+   * This function is created after simplifying the code of self::getRecipients(..) and these are the changes made
+   *  1. Takes mailing ID as the only parameter, and doesn't depend on mailing Job
+   *  2. This function ONLY fetch and store mailing recipients, doesn't handle SMS mode
+   *  3. Use less JOINs in underlying queries.
+   *  4. Simplifies code to fetch contacts from selected mailing smart group(s).
+   *  5. Doesn't execute query needlessly, e.g.
+   *    a) if SG (smart groups) are selected then ONLY queries resposible to fetch contacts from SG are executed
+   *    b) If no 'Include' groups are selected then simply return as this means no groups are selected to fetch recipients.
+   *  6. Refreshes smart group contact cache at once.
+   *
+   * @param CRM_Mailing_DAO_Mailing $mailingObj
    *
-   * @return CRM_Mailing_Event_BAO_Queue|string
+   * @return void
    */
-  public static function getRecipients(
-    $job_id,
-    $mailing_id = NULL,
-    $storeRecipients = FALSE,
-    $dedupeEmail = FALSE,
-    $mode = NULL) {
+  public static function getMailRecipients($mailingObj) {
```

Comment:
Could you put into function signature DocBlock an explanation of what $mailingObj is, eg the current mailing object (as opposed to a former excluded mailing object)?